src/net/sourceforge/kolmafia/swingui/panel/GenericPanel.java

```diff
@@ -171,26 +171,26 @@ public void addListeners() {
 
   private void addListener(final Object component, final ActionConfirmListener listener) {
     if (this.listenerMap == null) {
-      this.listenerMap = new HashMap();
+      this.listenerMap = new HashMap<>();
     }
 
     if (component instanceof JTextField) {
       ((JTextField) component).addKeyListener(listener);
-      this.listenerMap.put(component, new WeakReference(listener));
+      this.listenerMap.put(component, new WeakReference<>(listener));
     }
 
     if (component instanceof AutoFilterComboBox) {
       JTextComponent editor =
           (JTextComponent) ((AutoFilterComboBox) component).getEditor().getEditorComponent();
 
       editor.addKeyListener(listener);
-      this.listenerMap.put(editor, new WeakReference(listener));
-    } else if (component instanceof JComboBox && ((JComboBox) component).isEditable()) {
+      this.listenerMap.put(editor, new WeakReference<>(listener));
+    } else if (component instanceof JComboBox && ((JComboBox<?>) component).isEditable()) {
       JTextComponent editor =
-          (JTextComponent) ((JComboBox) component).getEditor().getEditorComponent();
+          (JTextComponent) ((JComboBox<?>) component).getEditor().getEditorComponent();
```

Comment:
(I would like to weaken this to just Component editor = ..getEditorComponent() since addKeyListener doesn't depend on those concrete details, but that's irrelevant to this change.)